src/lib/network_pool/batcher.ml

```diff
@@ -1,68 +1,36 @@
 open Core_kernel
 open Async_kernel
 open Network_peer
-open Pickles_types
 
 module Id = Unique_id.Int ()
 
-module Outcome = struct
-  (* The final result of a batch verification. For each proof, a determination of
-     whether or not it is valid.
-
-     We allow valid proofs to carry data of type 'result, rather than just being
-     propositional. This is to support, e.g., User_command.t's getting
-     verified into a result of type User_command.Valid.t.
-  *)
-  type ('proof, 'result) t = [`Valid of 'result | `Invalid of 'proof] Id.Map.t
-  [@@deriving sexp]
-
-  let invalid invalid =
-    Id.Map.of_alist_exn (List.map invalid ~f:(fun (id, x) -> (id, `Invalid x)))
-
-  let to_invalid (t : _ t) =
-    Map.fold t ~init:[] ~f:(fun ~key:_ ~data acc ->
-        match data with `Valid _ -> acc | `Invalid x -> x :: acc )
-
-  let of_alist_exn : _ -> _ t = Id.Map.of_alist_exn
-
-  let empty : _ t = Id.Map.empty
-
-  (* Merge the outcomes for two batches. Used to recombine results of
-     recursive splitting *)
-  let append (type p r) (t1 : (p, r) t) (t2 : (p, r) t) : (p, r) t =
-    Id.Map.merge t1 t2 ~f:(fun ~key:_ d ->
-        match d with
-        | `Left x | `Right x ->
-            Some x
-        | `Both (x, y) -> (
-          (* This should never happen. Still, we handle it
-          in case the truly unthinkable occurs *)
-          match (x, y) with
-          | `Valid _, `Valid _ ->
-              Some x
-          | `Invalid _, `Invalid _ ->
-              Some x
-          | `Valid _, `Invalid _ | `Invalid _, `Valid _ ->
-              assert false ) )
-end
+type ('init, 'result) elt =
+  { id: Id.t
+  ; data: 'init
+  ; weight: int
+  ; res: ('result, unit) Result.t Or_error.t Ivar.t sexp_opaque }
```

Comment:
This always seems to be filled by an `Ok _`; can we get rid of the `Or_error.t` here?

Reply:
Good catch -- I think this is actually an error on my part, and there should be an error case. Let me remedy that.